### PR TITLE
Registry migration tokens map refactor

### PR DESCRIPTION
**Pull request by ug02fast**
Removing tokens_map and dependencies of it.

branched off swap refactor. swap refactor should go first.

tested swapping locally.

### CODE REVIEW

**Review by nguyenaiden — APPROVED** [review]
+1 on this. Also, reiterating to move towards default state to reduce null/undefined checks

**Comment by hammeiam** [comment]
fyi you can specify your root branch. So if your branches are `master -> A -> B`, you can set your root to A instead of master and github will only display the diff between A and B. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

File: src/utils/updateTokenPrices.ts
Comment (hammeiam) [inline src/utils/updateTokenPrices.ts:28]: I think it would be better to move the changes to this file + App.ts to another PR, so we can revert just that if something ends up going wrong once the new price fetching logic is engaged.

File: src/utils/__tests__/index.ts
Comment (hammeiam) [inline src/utils/__tests__/index.ts:197]: rm console logs

File: src/constants/index.ts
Comment (hammeiam) [inline src/constants/index.ts:2063]: amaaaaaazing

File: src/hooks/useContract.ts
Comment (hammeiam) [inline src/hooks/useContract.ts:343]: Can you remove this function too? It's only used in `useApproveAndSwap`, and only to get erc20 contracts for approval.

File: src/pages/App.tsx
Comment (hammeiam) [inline src/pages/App.tsx:203]: mentioned elsewhere, let's save turning this on for another PR

File: src/hooks/usePendingSwapData.ts
Comment (hammeiam) [inline src/hooks/usePendingSwapData.ts:525]: it's okay to return null here if either of these tokens are undefined. Just make sure to `console.error` it. This means that you no longer have to add `undefined` to the `PendingSwap` type

File: src/hooks/useTokenMaps.ts
Comment (hammeiam) [inline src/hooks/useTokenMaps.ts:7]: You can simplify this whole file once https://github.com/saddle-finance/saddle-frontend/pull/1193 lands

File: src/hooks/useTokenMaps.ts
Comment (hammeiam) [inline src/hooks/useTokenMaps.ts:15]: `if !pool return acc`, get rid of `pool?.`

File: src/hooks/useTokenMaps.ts
Comment (hammeiam) [inline src/hooks/useTokenMaps.ts:35]: kill PoolName

File: src/hooks/useTokenMaps.ts
Comment (hammeiam) [inline src/hooks/useTokenMaps.ts:32]: `if !token return`, skip all the `token?.symbol ?? ""` later

File: src/pages/Swap.tsx
Comment (hammeiam) [inline src/pages/Swap.tsx:131]: you can remove `address` here, the old logic was just making sure the token existed on the current chain, but `tokens` does that automatically.

File: src/pages/Swap.tsx
Comment (hammeiam) [inline src/pages/Swap.tsx:165]:
```
const token = tokensMap?.[to.symbol]
if (!token) log and return null
```
now token is guaranteed to be defined and you can skip all of the `token?.name ?? ""` again

File: src/pages/Swap.tsx
Comment (hammeiam) [inline src/pages/Swap.tsx:258]: this will fail if we don't know the address. Write to the `error` var and set `amountToReceive, amountMediumSynth` to `Zero` instead

File: src/pages/Swap.tsx
Comment (hammeiam) [inline src/pages/Swap.tsx:274]: same as above

File: src/pages/Swap.tsx
Comment (hammeiam) [inline src/pages/Swap.tsx:287]: same as above

File: src/pages/Swap.tsx
Comment (hammeiam) [inline src/pages/Swap.tsx:239]: hoist both of these to the top of the function. I think we should either assume that if these tokens are in `formStateArg`, then they must also be in `tokensMap`, then we can force them to be `BasicToken` and not undefined. Or we check if they're undefined around line 213 and not even try to use them

File: src/pages/Swap.tsx
Comment (hammeiam) [inline src/pages/Swap.tsx:335]: Prefer checking for token existence instead of making up numbers

File: src/pages/Swap.tsx
Comment (hammeiam) [inline src/pages/Swap.tsx:586]: same as above, prefer not to make up numbers.

File: src/hooks/usePendingSwapData.ts
Comment (hammeiam) [inline src/hooks/usePendingSwapData.ts:26]: see comment below

File: src/components/PendingSwapExchange.tsx
Comment (hammeiam) [inline src/components/PendingSwapExchange.tsx:41]: these changes are all moot if you don't add `| undefined` to `PendingSwap` type, see comments in `usePendingSwapData`

File: src/components/PendingSwapModal.tsx
Comment (hammeiam) [inline src/components/PendingSwapModal.tsx:197]: same as `PendingSwapExchange` I believe

File: src/components/PendingSwapTimeline.tsx
Comment (hammeiam) [inline src/components/PendingSwapTimeline.tsx:44]: same as `PendingSwapExchange`

File: src/components/SwapPage.tsx
Comment (hammeiam) [inline src/components/SwapPage.tsx:262]: same as `PendingSwapExchange`

**Review by hammeiam — CHANGES_REQUESTED** [review]
tldr assert that a token exists once by early returning, save yourself from having to write `token?.symbol ?? ""` in 8 different files

File: src/hooks/usePendingSwapData.ts
Comment (nguyenaiden) [inline src/hooks/usePendingSwapData.ts:341]: While you're at it, could you take a look and change these to
```
const events = (
    await Promise.all(
      eventFilters.map((filter) =>
        bridgeContract.queryFilter(filter, startBlock, "latest"),
      ),
    )
  ).flat()

  // Step 3: fetch aync data via bridge contract
  const fetchedPendingSwaps = (
    await Promise.all(
      events.map((event) =>
        fetchPendingSwapInfo(bridgeContract, event, chainId),
      ),
    )
  ).filter(Boolean)
  if (fetchedPendingSwaps.length === 0) return
  ```
Reassignments are usually not a good idea.

Comment (ug02fast) [reply src/hooks/usePendingSwapData.ts:341]: This is a good call but can you make an issue on this and make the change in a separate PR for the sake of keeping the scope of this to removal of POOL/TOKENS_MAP?

File: src/utils/updateTokenPrices.ts
Comment (ug02fast) [reply src/utils/updateTokenPrices.ts:28]: Token price fetching is working with the changes in this PR. What do you think would be a reason for something to go wrong?

File: src/pages/Swap.tsx
Comment (ug02fast) [reply src/pages/Swap.tsx:609]: wasnt quiet able to figure this one out.

**Review by nguyenaiden — APPROVED** [review]
Okay this looks good to continue with the migration. I've manually tested deposit, swap, and withdraw locally.

Good to merge after addressing a couple of comments

Will need to create tickets to go back and address setting default states to allow early exits and remove optional. High priority

File: src/hooks/useCalculateSwapPairs.ts
Comment (ug02fast) [reply src/hooks/useCalculateSwapPairs.ts:106]: done in following PR https://github.com/saddle-finance/saddle-frontend/pull/1182/files#diff-ab89847bbe63169d79f42de9dbaaac7dd80b8651f8bd33a3a4e7f523d34c191bR106

File: src/hooks/useCalculateSwapPairs.ts
Comment (ug02fast) [reply src/hooks/useCalculateSwapPairs.ts:103]: For the sake of keeping the scope of PR from pulling out TOKENS_MAP going to leave as is. It would be better to discuss conventions as a team and make one PR to replace everywhere.

File: src/hooks/usePendingSwapData.ts
Comment (ug02fast) [reply src/hooks/usePendingSwapData.ts:507]: For the sake of keeping the scope of PR from pulling out TOKENS_MAP going to leave as is. It would be better to discuss conventions as a team and make one PR to replace everywhere.

File: src/hooks/useApproveAndSwap.ts
Comment (hammeiam) [inline src/hooks/useApproveAndSwap.ts:84]: Awesome! This is so much better. Nit, no need to lowercase address here. The only reason we do that elsewhere is to make addresses uniform for equality checks

**Pull request merged**
